plaidml2/op/lib/ops.cc

```diff
@@ -1957,17 +1957,49 @@ Value reshape(const Value& value) {
   if (args.size() != 2) {
     throw std::runtime_error(str(boost::format("PlaidML reshape op expects 2 arguments (received %1%)") % args.size()));
   }
+
   auto I = args[0].as_tensor();
   std::vector<TensorDim> dims;
+  int64_t dim_counter = 0;
+  int64_t neg_idx = -1;
+  std::vector<TensorDim> I_dims(I.shape().ndims());
+  I.bind_dims(I_dims);
+
   for (auto dim : args[1].as_tuple()) {
     if (dim.is_int()) {
       dims.emplace_back(dim.as_int());
     } else if (dim.is_dim()) {
       dims.emplace_back(dim.as_dim());
     } else if (dim.is_str()) {
       // TODO: handle special cases
-      dims.emplace_back(0);
+      // TODO: figure out why there are quotation marks around fill and match
+      if (strcmp("\"match\"", dim.str().c_str()) == 0) {
```

Comment:
We should never use `strcmp`, it's inherently unsafe (and might be banned from automated static analysis tools).